tests/blackbox_test.go

```diff
@@ -79,15 +91,35 @@ func TestMain(m *testing.M) {
 func TestIgnitionBlackBox(t *testing.T) {
 	for _, test := range register.Tests[register.PositiveTest] {
 		test := test
+		var uConfig IgnitionConfig
+		if test.ConfigVersion != "3.1.0" && test.ConfigVersion != "3.0.0" && test.Config != "" {
+			if err := json.Unmarshal([]byte(test.Config), &uConfig); err != nil {
+				t.Log(test.ConfigVersion)
+				t.Log(uConfig)
+				t.Error(fmt.Errorf("error parsing Ignition configuration: %v", err))
+			}
+		}
 		t.Run(test.Name, func(t *testing.T) {
+			defer func() {
+				for _, luks := range uConfig.Storage.Luks {
+					t.Log(luks.Name)
+					if err := removeLuksDevice(luks.Name); err != nil {
+						t.Error(fmt.Errorf("failed to remove existing LUKS device %s: %v", luks.Name, err))
+					}
+				}
+			}()
```

Comment:
Instead of this, I would expect this stuff to live in `outer()` like the other things. See e.g. https://github.com/coreos/ignition/blob/32221a98e6d1be16e8b6bc39f65424b696021c22/tests/blackbox_test.go#L197-L206 where we defer removing the loopback device and deleting it. This would be another `defer` after those (so it runs before them on the way out).

Reply:
Ok awesome, I just relised 'defer' operates like a stack... thank you, that was likely my issue before in the outer fnc...